test_view_build_status_with_replace is flaky

Appears to time out.

```
Error Message
AssertionError: Deadline exceeded, failing test.
Stacktrace
manager = <test.pylib.manager_client.ManagerClient object at 0x7f1fbff255e0>

    @pytest.mark.asyncio
    async def test_view_build_status_with_replace_node(manager: ManagerClient):
        node_count = 4
        servers = await manager.servers_add(node_count)
        cql, _ = await manager.get_ready_cql(servers)
    
        await create_keyspace(cql)
        await create_table(cql)
        await create_mv(cql, "vt1")
        await create_mv(cql, "vt2")
    
        await wait_for_view(cql, "vt1", node_count)
>       await wait_for_view(cql, "vt2", node_count)

test/topology_custom/test_view_build_status.py:394: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/pylib/util.py:248: in wait_for_view
    await wait_for(view_is_built, deadline)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pred = <function wait_for_view.<locals>.view_is_built at 0x7f1fbff0d6c0>
deadline = 1726497149.690381, period = 1, before_retry = None

    async def wait_for(
            pred: Callable[[], Awaitable[Optional[T]]],
            deadline: float,
            period: float = 1,
            before_retry: Optional[Callable[[], Any]] = None) -> T:
        while True:
>           assert(time.time() < deadline), "Deadline exceeded, failing test."
E           AssertionError: Deadline exceeded, failing test.

test/pylib/util.py:54: AssertionError
```

Seen again in CI run: https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/12899/testReport/junit/(root)/test_view_build_status/Tests___Unit_Tests___test_view_build_status_with_replace_node_release_1/
```
manager = <test.pylib.manager_client.ManagerClient object at 0x7efc3c6a6630>

    @pytest.mark.asyncio
    async def test_view_build_status_with_replace_node(manager: ManagerClient):
        node_count = 4
        servers = await manager.servers_add(node_count)
        cql, _ = await manager.get_ready_cql(servers)
    
        await create_keyspace(cql)
        await create_table(cql)
        await create_mv(cql, "vt1")
        await create_mv(cql, "vt2")
    
        await wait_for_view(cql, "vt1", node_count)
>       await wait_for_view(cql, "vt2", node_count)

test/topology_custom/test_view_build_status.py:394: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
test/pylib/util.py:249: in wait_for_view
    await wait_for(view_is_built, deadline)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pred = <function wait_for_view.<locals>.view_is_built at 0x7efc3c6f84a0>
deadline = 1730387157.868799, period = 1, before_retry = None

    async def wait_for(
            pred: Callable[[], Awaitable[Optional[T]]],
            deadline: float,
            period: float = 1,
            before_retry: Optional[Callable[[], Any]] = None) -> T:
        while True:
>           assert(time.time() < deadline), "Deadline exceeded, failing test."
E           AssertionError: Deadline exceeded, failing test.
```

--- a/db/view/view.cc
+++ b/db/view/view.cc
@@ -2448,11 +2448,11 @@ view_builder::view_build_statuses(sstring keyspace, sstring view_name) const {
 
 future<> view_builder::add_new_view(view_ptr view, build_step& step) {
     vlogger.info0("Building view {}.{}, starting at token {}", view->ks_name(), view->cf_name(), step.current_token());
+    if (this_shard_id() == 0) {
+        co_await mark_view_build_started(view->ks_name(), view->cf_name());
+    }
+    co_await _sys_ks.register_view_for_building(view->ks_name(), view->cf_name(), step.current_token());
     step.build_status.emplace(step.build_status.begin(), view_build_status{view, step.current_token(), std::nullopt});
-    auto f = this_shard_id() == 0 ? mark_view_build_started(view->ks_name(), view->cf_name()) : make_ready_future<>();
-    return when_all_succeed(
-            std::move(f),
-            _sys_ks.register_view_for_building(view->ks_name(), view->cf_name(), step.current_token())).discard_result();
 }
 
 static future<> flush_base(lw_shared_ptr<replica::column_family> base, abort_source& as) {

--- a/db/view/view.cc
+++ b/db/view/view.cc
@@ -2356,7 +2356,7 @@ static future<> announce_with_raft(
 
 future<> view_builder::mark_view_build_started(sstring ks_name, sstring view_name) {
     co_await write_view_build_status(
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             co_await utils::get_local_injector().inject("view_builder_pause_add_new_view", utils::wait_for_message(5min));
             const sstring query_string = format("INSERT INTO {}.{} (keyspace_name, view_name, host_id, status) VALUES (?, ?, ?, ?)",
                     db::system_keyspace::NAME, db::system_keyspace::VIEW_BUILD_STATUS_V2);
@@ -2365,7 +2365,7 @@ future<> view_builder::mark_view_build_started(sstring ks_name, sstring view_nam
                     {std::move(ks_name), std::move(view_name), host_id.uuid(), "STARTED"},
                     "view builder: mark view build STARTED");
         },
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             co_await utils::get_local_injector().inject("view_builder_pause_add_new_view", utils::wait_for_message(5min));
             co_await _sys_dist_ks.start_view_build(std::move(ks_name), std::move(view_name));
         }
@@ -2374,7 +2374,7 @@ future<> view_builder::mark_view_build_started(sstring ks_name, sstring view_nam
 
 future<> view_builder::mark_view_build_success(sstring ks_name, sstring view_name) {
     co_await write_view_build_status(
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             co_await utils::get_local_injector().inject("view_builder_pause_mark_success", utils::wait_for_message(5min));
             const sstring query_string = format("UPDATE {}.{} SET status = ? WHERE keyspace_name = ? AND view_name = ? AND host_id = ?",
                     db::system_keyspace::NAME, db::system_keyspace::VIEW_BUILD_STATUS_V2);
@@ -2383,7 +2383,7 @@ future<> view_builder::mark_view_build_success(sstring ks_name, sstring view_nam
                     {"SUCCESS", std::move(ks_name), std::move(view_name), host_id.uuid()},
                     "view builder: mark view build SUCCESS");
         },
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             co_await utils::get_local_injector().inject("view_builder_pause_mark_success", utils::wait_for_message(5min));
             co_await _sys_dist_ks.finish_view_build(std::move(ks_name), std::move(view_name));
         }
@@ -2392,14 +2392,14 @@ future<> view_builder::mark_view_build_success(sstring ks_name, sstring view_nam
 
 future<> view_builder::remove_view_build_status(sstring ks_name, sstring view_name) {
     co_await write_view_build_status(
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             const sstring query_string = format("DELETE FROM {}.{} WHERE keyspace_name = ? AND view_name = ?",
                     db::system_keyspace::NAME, db::system_keyspace::VIEW_BUILD_STATUS_V2);
             co_await announce_with_raft(_qp, _group0_client, _as, std::move(query_string),
                     {std::move(ks_name), std::move(view_name)},
                     "view builder: delete view build status");
         },
-        [&] () -> future<> {
+        [this, ks_name, view_name] () -> future<> {
             co_await _sys_dist_ks.remove_view(std::move(ks_name), std::move(view_name));
         }
     );

--- a/db/view/view.cc
+++ b/db/view/view.cc
@@ -2050,7 +2050,6 @@ future<> view_builder::start_in_background(service::migration_manager& mm, utils
         // the view build information.
         fail.cancel();
         co_await barrier.arrive_and_wait();
-        units.return_all();
 
         _mnotifier.register_listener(this);
         co_await calculate_shard_build_step(vbi);

